pelican/plugins/show_source/show_source.py

```diff
@@ -1,59 +1,78 @@
 import logging
 import os
-from urllib.parse import urljoin
+import six
 
 from pelican import signals
 from pelican.utils import pelican_open
+from six.moves.urllib.parse import urljoin
+
+if not six.PY3:
+    from codecs import open
 
 logger = logging.getLogger(__name__)
 source_files = []
-PROCESS = ["articles", "pages", "drafts"]
-
+TYPES_TO_PROCESS = ['articles', 'pages']
 
 def link_source_files(generator):
     """
     Processes each article/page object and formulates copy from and copy
     to destinations, as well as adding a source file URL as an attribute.
     """
     # Get all attributes from the generator that are articles or pages
-    posts = [
+    documents = sum([
         getattr(generator, attr, None)
-        for attr in PROCESS
-        if getattr(generator, attr, None) is not None
-    ]
+        for attr in TYPES_TO_PROCESS
+        if getattr(generator, attr, None)
+    ], [])
+
+    autoext_setting = generator.settings.get(
+        'SHOW_SOURCE_AUTOEXT', False
+    )
+
     # Work on each item
-    for post in posts[0]:
-        if (
-            "SHOW_SOURCE_ON_SIDEBAR" not in generator.settings
-            and "SHOW_SOURCE_IN_SECTION" not in generator.settings
-        ):
+    for post in documents:
+        if not ('SHOW_SOURCE_ON_SIDEBAR' in generator.settings or 'SHOW_SOURCE_IN_SECTION' in generator.settings):
             return
+
         # Only try this when specified in metadata or SHOW_SOURCE_ALL_POSTS
         # override is present in settings
-        if (
-            "SHOW_SOURCE_ALL_POSTS" in generator.settings
-            or "show_source" in post.metadata
-        ):
+        if 'SHOW_SOURCE_ALL_POSTS' in generator.settings or 'show_source' in post.metadata:
             # Source file name can be optionally set in config
             show_source_filename = generator.settings.get(
-                "SHOW_SOURCE_FILENAME", "{}.txt".format(post.slug)
+                'SHOW_SOURCE_FILENAME', '{}.txt'.format(post.slug)
             )
             try:
                 # Get the full path to the original source file
-                source_out = os.path.join(post.settings["OUTPUT_PATH"], post.save_as)
+                source_out = os.path.join(post.settings['OUTPUT_PATH'], post.save_as)
+
                 # Get the path to the original source file
                 source_out_path = os.path.split(source_out)[0]
+
                 # Create 'copy to' destination for writing later
                 copy_to = os.path.join(source_out_path, show_source_filename)
+
                 # Add file to published path
                 source_url = urljoin(post.save_as, show_source_filename)
             except Exception:
-                return
+                logger.error("Error processing source file for post", exc_info=True)
```

Comment:
I suggest prefixing the error message with the name of the plugin. Otherwise, it might be hard for users to know from where the error originated.

Reply:
I believe the logger is already namespaced by this point with `logging.getLogger(__name__)`? I'll run a debug round to be sure the logger is configured to print its name.

Reply:
Oh, you might be right. Please check and let me know what you find.

Reply:
The logger is internally namespaced but pelican's log formatter doesn't print that information. For now I'm editing it as suggested